pkg/registry/interface.go

```diff
@@ -0,0 +1,23 @@
+package registry
+
+import (
+	"context"
+
+	"github.com/notaryproject/notation-go"
+	"github.com/opencontainers/go-digest"
+)
+
+// SignatureRepository provides a storage for signatures
+type SignatureRepository interface {
```

Comment:
Should we have a method for `GetSignatureManifest(ctx context.Context, manifestDescriptor notation.Descriptor)` to return contents of a signature manifest?

Reply:
What's the intention of this? Since it is an interface, we want to decouple it with specific manifest structures.

Reply:
The intention is to get the signature manifest's annotations (`io.cncf.notary.x509chain.thumbprint#S256`) and use them for signature verification.